src/v3/Tracer.ts

```diff
@@ -200,41 +208,62 @@ export class Tracer<
       RelationSchemasT,
       VariantsT
     >,
+    opts?: {
+      noDraftPresentBehavior: 'error' | 'warning' | 'noop'
+    },
   ): void => {
     const trace:
       | Trace<SelectedRelationNameT, RelationSchemasT, VariantsT>
       | undefined = this.traceUtilities.getCurrentTrace()
 
+    let reportingFunction
+    const { noDraftPresentBehavior } = opts ?? {}
+
+    switch (noDraftPresentBehavior) {
+      case 'error':
+        reportingFunction = this.traceUtilities.reportErrorFn
+        break
+      case 'noop':
+        reportingFunction = undefined
+        break
+      default:
+        reportingFunction = this.traceUtilities.reportWarningFn
+    }
+
     if (!trace) {
-      this.traceUtilities.reportErrorFn(
+      reportingFunction?.(
         new Error(
           `No currently active trace when initializing a trace. Call tracer.start(...) or tracer.createDraft(...) beforehand.`,
         ),
+        // @ts-expect-error TS doesn't like this type for some reason
+        { definition: this.definition },
       )
       return
     }
 
     // this is an already initialized active trace, do nothing:
     if (!trace.isDraft) {
-      this.traceUtilities.reportWarningFn(
+      reportingFunction?.(
         new Error(
           `You are trying to initialize a trace that has already been initialized before (${trace.definition.name}).`,
         ),
+        // @ts-expect-error TS doesn't like this type for some reason
+        { definition: this.definition },
```

Comment:
```suggestion
        trace,
```

should probably work